### PR TITLE
Cairo0 unsafe keccak finalize

### PR DESCRIPTION
**Pull request by MaksymMalicki**

### CODE REVIEW

**Comment by MaksymMalicki** [comment]
The same situation appears here, as [in this comment](https://github.com/NethermindEth/cairo-vm-go/pull/361#issuecomment-2108288726)

**Comment by TAdev0** [comment]
Closes #287

File: pkg/hintrunner/zero/zerohint_keccak_test.go [hunk 1]
```diff
@@ -324,6 +325,95 @@ func TestZeroHintKeccak(t *testing.T) {
...
+				},
+			},
+			{
+				// end - start is 0
+				operanders: []*hintOperander{
+					{Name: "keccak_state.start_ptr", Kind: apRelative, Value: addr(6)},
+					{Name: "keccak_state.end_ptr", Kind: apRelative, Value: addr(9)},
+					{Name: "keccak_state.0", Kind: apRelative, Value: new(fp.Element).Sub(&utils.FeltMax128, &utils.FeltOne)},
+					{Name: "keccak_state.1", Kind: apRelative, Value: new(fp.Element).Sub(&utils.FeltMax128, &utils.FeltOne)},
+					{Name: "keccak_state.2", Kind: apRelative, Value: new(fp.Element).Sub(&utils.FeltMax128, &utils.FeltOne)},
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_keccak_test.go:388]: what is `keccak_state.2` for ?

Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_keccak_test.go:388]: why not using addr(8) for end_ptr and remove keccak_state.2 ?

**Review by TAdev0 — APPROVED** [review]

File: pkg/hintrunner/zero/zerohint_keccak.go [hunk 2]
```diff
@@ -172,6 +192,113 @@ func createUnsafeKeccakHinter(resolver hintReferenceResolver) (hinter.Hinter, er
 	return newUnsafeKeccakHint(data, length, high, low), nil
 }
 
+// UnsafeKeccakFinalize computes keccak hash of the data in memory without validity enforcement and writes the result in the `low` and `high` memory cells. It gets the data pointers from the `keccakState` memory cell, computing the hash of the data in the range [start_ptr, end_ptr).
+//
+// `newUnsafeKeccakFinalizeHint` takes 3 operanders as arguments
+//   - `keccakState` is the address in memory where the data pointers are stored
+//   - `low` is the low part of the produced hash
```
[the remark this answers is not in the document]
Comment (MaksymMalicki) [reply pkg/hintrunner/zero/zerohint_keccak.go:199]: can be 😁

File: pkg/hintrunner/zero/zerohint_keccak_test.go [hunk 3]
```diff
@@ -324,6 +325,111 @@ func TestZeroHintKeccak(t *testing.T) {
...
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newUnsafeKeccakFinalizeHint(ctx.operanders["keccak_state.start_ptr"], ctx.operanders["high"], ctx.operanders["low"])
+				},
+				check: func(t *testing.T, ctx *hintTestContext) {
+					varValueEquals("high", feltString("138808390298063253082378171006150548014"))(t, ctx)
+					varValueEquals("low", feltString("105025107803124291085217464954131521934"))(t, ctx)
+				},
+			},
+			{
+				// end - random big values
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_keccak_test.go:382]: end - random big values? what does it mean? :p

File: pkg/hintrunner/zero/zerohint_keccak_test.go (hunk 1 above)
Comment (MaksymMalicki) [reply pkg/hintrunner/zero/zerohint_keccak_test.go:388]: Changed the name of input values to input.x.
Removing the third element of input would change the output

**Review by har777 — APPROVED** [review]

**Pull request merged**